django_fido/js/fido2.js

```diff
@@ -124,18 +124,20 @@ async function startFido2() {
     if (!form)
         return // Silently skip if not on correct page.
 
+    if (!isFido2Availabile()) {
+        addFido2Error(TRANSLATIONS.FIDO2_NOT_AVAILABLE)
+        return
+    }
+
+    if (form.dataset.mode === FIDO2_AUTHENTICATION_REQUEST)
+        await sendFido2AuthenticationRequest(form.dataset.url)
+
     const submit_button = document.getElementById('submit-button')
     if (submit_button) {
         submit_button.addEventListener('click', async e => {
             e.preventDefault()
 
-            if (!isFido2Availabile()) {
-                addFido2Error(TRANSLATIONS.FIDO2_NOT_AVAILABLE)
-                return
-            }
-            if (form.dataset.mode === FIDO2_AUTHENTICATION_REQUEST) {
-                await sendFido2AuthenticationRequest(form.dataset.url)
-            } else if (form.dataset.mode === FIDO2_REGISTRATION_REQUEST) {
+            if (form.dataset.mode === FIDO2_REGISTRATION_REQUEST) {
```

Comment:
This condition could be much higher now, above the submit button detection.

Reply:
Yes, it will be better. Fixed.

Reply:
ok